pyteal/ast/itxn.py

```diff
@@ -158,33 +168,42 @@ def SetField(cls, field: TxnField, value: Expr | list[Expr]) -> Expr:
                 compatible with the field being set.
         """
         if not field.is_array:
-            if type(value) is list:
+            if type(value) is list or isinstance(value, TxnArray):
                 raise TealInputError(
                     "inner transaction set field {} does not support array value".format(
                         field
                     )
                 )
             return InnerTxnFieldExpr(field, cast(Expr, value))
         else:
-            if type(value) is not list:
+            if type(value) is not list and not isinstance(value, TxnArray):
                 raise TealInputError(
                     "inner transaction set array field {} with non-array value".format(
                         field
                     )
                 )
-            for valueIter in value:
-                if not isinstance(valueIter, Expr):
-                    raise TealInputError(
-                        "inner transaction set array field {} with non PyTeal expression array element {}".format(
-                            field, valueIter
+
+            if type(value) is list:
+                for valueIter in value:
+                    if not isinstance(valueIter, Expr):
+                        raise TealInputError(
+                            "inner transaction set array field {} with non PyTeal expression array element {}".format(
+                                field, valueIter
+                            )
                         )
-                    )
-            return Seq(
-                *[
-                    InnerTxnFieldExpr(field, cast(Expr, valueIter))
-                    for valueIter in value
-                ]
-            )
+
+                return Seq(
+                    *[
+                        InnerTxnFieldExpr(field, cast(Expr, valueIter))
+                        for valueIter in value
+                    ]
+                )
+            else:
+                return For(
+                    (i := ScratchVar()).store(Int(0)),
+                    i.load() < cast(TxnArray, value).length(),
+                    i.store(i.load() + Int(1)),
+                ).Do(InnerTxnFieldExpr(field, i.load()))
```

Comment:
hold up, these lines seem off: I assume you want to put everything in `value`, which is a `TxnArray`, in `itxn_field` setting. But it looks like you are setting 0,1,2... to itxn_field, which I am not sure if it is making sense.

Reply:
i'll need to actually test it against the avm but i _think_ you can reference the relevant array by index and it knows where to look. I'm not positive though

Reply:
> algosdk.error.AlgodHTTPError: 169: itxn_field Accounts arg 0 wanted type []byte got uint64

no worky :(

Reply:
yikes